src/query/query_parse.rs

```diff
@@ -0,0 +1,131 @@
+use crate::query::query_types::*;
+use pest::Parser;
+use crate::query::QueryError;
+
+#[derive(Parser)]
+#[grammar = "query/query.pest"]
+struct QueryParser;
+
+pub fn parse_query(query: &str) -> Result<QueryNode, QueryError> {
+    match QueryParser::parse(Rule::query, query) {
+        Ok(pairs) => {
+            // Below never fails as query has been parsed and we know query rule can only have 1
+            // expr and its inner can only be length 1 also (either latest or a string)
+            let query = pairs.peek().unwrap().into_inner().peek().unwrap();
+            parse_query_content(query)
+        }
+        Err(e) => Err(QueryError::ParseError(Box::new(e))),
+    }
+}
+
+fn parse_query_content(query: pest::iterators::Pair<Rule>) -> Result<QueryNode, QueryError> {
+    match query.as_rule() {
+        Rule::string => {
+            let x = get_string_inner(query);
+            Ok(QueryNode::Lookup(LookupLhs::Id, x))
+        }
+        Rule::noVariableFunc => {
+            // Pest asserts for us that the only possible no variable func is latest()
+            // we might want to move this validation into Rust code later to return
+            // better errors to the user
+            Ok(QueryNode::Latest(None))
+        }
+        Rule::infixExpression => {
+            let mut infix = query.into_inner();
+            let first_arg = infix.next().unwrap();
```

Comment:
lots of `unwrap` here, which I feel like is something we're supposed to avoid. how about `.map(|e| e.as_str())` instead and then use `Some("id")` etc on line 41 etc? and similarly for other options?

Reply:
In the pest docs (https://pest.rs/book/parser_api.html#using-pair-and-pairs-with-a-grammar) it says 

> you should rely on the meaning of your grammar for properties such as "contains n sub-rules", "is safe to parse to f32", and "never fails to match". Idiomatic pest code uses unwrap and unreachable!.

I can update the parsing to remove usages of unwrap but I think they are safe in the parsing because `pest` ensures that those things are available. I'm happy update this to not use `unwrap` if you feel strongly but I think it is not as bad in this context as it is in other code.